### PR TITLE
Fix bug with https proxy acquired cleanup #1340

### PR DESCRIPTION
**Pull request by dodge-ksv**
## What do these changes do?
This pull request fix bug with https proxy acquired cleanup. 
When TCPConnector creates a proxy connection, it adds a transport to the acquired and then sends CONNECT request. But after that TCPConnector detach the transport and does not remove them from the acquired. This caused a memory leak when https proxy connection had used.

## Are there changes in behavior for the user?
No.

## Related issue number
#1340

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [x] Add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names. 
- [x] Add a new entry to `CHANGES.rst`
  * Choose any open position to avoid merge conflicts with other PRs.
  * Add a link to the issue you are fixing (if any) using `#issue_number` format at the end of changelog message. Use Pull Request number if there are no issues for PR or PR covers the issue only partially.

### CODE REVIEW

File: tests/test_proxy_functional.py [hunk 1]
```diff
@@ -169,6 +181,84 @@ def test_proxy_http_auth_from_url(proxy_test_server, get_request):
...
+        assert 1 == len(conn._acquired.keys())
+        assert 1 == len(conn._acquired[key])
+
+        yield from resp.release()
+
+    yield from request()
+
+    assert 0 == len(conn._acquired[key])
+
+    sess.close()
```
[the remark this answers is not in the document]
Comment (asvetlov) [reply tests/test_proxy_functional.py]: `yield from sess.close()` is enough, session takes ownership of connector.

**Comment by asvetlov** [comment]
Please take a look on Windows failed tests

**Comment by kserhii** [comment]
Tests fails with the message:
     OSError: [WinError 10038] An operation was attempted on something that is not a socket
but I can't understand why this happened.
Maybe you have an idea?

**Comment by asvetlov** [comment]
Maybe you do mocking too much or not enough? :)

**Comment by kserhii** [comment]
The problem was with unclosed proxy transport. Now everything is done.

**Comment by asvetlov** [comment]
Perfect!

**Pull request merged**